spec/helpers/search_helper_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'spec_helper'
+
+RSpec.describe SearchHelper, type: :helper do
+  def add_qualification_fixtures
+    Qualification.create!(id: 1, name: 'Qual 1', order: 1)   # Ignored
+    Qualification.create!(id: 2, name: 'Qual 2', order: 3)   # Not ignored
+    Accreditation.create!(id: 1, name: 'Accred 1', order: 4) # Ignored
+    Accreditation.create!(id: 2, name: 'Accred 2', order: 1) # Not ignored
+  end
+
+  describe '#firm_has_qualifications_or_accreditations?' do
+    before { add_qualification_fixtures }
+    let(:firm) { double(FirmResult, adviser_qualification_ids: adviser_qualification_ids, adviser_accreditation_ids: adviser_accreditation_ids) }
+    subject { helper.firm_has_qualifications_or_accreditations?(firm) }
+
+    NO_QUALIFICATIONS = []
+    NO_ACCREDITATIONS = []
+    IGNORED_QUALIFICATIONS = [1]
+    IGNORED_ACCREDITATIONS = [1]
+    VISIBLE_QUALIFICATIONS = [2]
+    VISIBLE_ACCREDITATIONS = [2]
+
+    QUALIFICATIONS_TRUTH_TABLE = [
+      [NO_QUALIFICATIONS,      NO_ACCREDITATIONS,      false],
+      [NO_QUALIFICATIONS,      IGNORED_ACCREDITATIONS, false],
+      [NO_QUALIFICATIONS,      VISIBLE_ACCREDITATIONS, true],
+      [IGNORED_QUALIFICATIONS, NO_ACCREDITATIONS,      false],
+      [IGNORED_QUALIFICATIONS, IGNORED_ACCREDITATIONS, false],
+      [IGNORED_QUALIFICATIONS, VISIBLE_ACCREDITATIONS, true],
+      [VISIBLE_QUALIFICATIONS, NO_ACCREDITATIONS,      true],
+      [VISIBLE_QUALIFICATIONS, IGNORED_ACCREDITATIONS, true],
+      [VISIBLE_QUALIFICATIONS, VISIBLE_ACCREDITATIONS, true]
+    ]
+
+    QUALIFICATIONS_TRUTH_TABLE.each.with_index do |row, index|
+      context "with truth table row #{index}" do
+        let(:adviser_qualification_ids) { row[0] }
+        let(:adviser_accreditation_ids) { row[1] }
+        it { is_expected.to be row[2] }
+      end
+    end
```

Comment:
Reading these tests in these format will help to understand very quick what's the code's behaviour :). Wdyt?